src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2131,6 +2134,65 @@ private void checkInternetPermission() {
         }
     }
 
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            /* the current serialize/deserialize feature is for first party 
+             * so the client ID for the FoCI token cache item is hard coded below
+             */
+            final String foci = "1";
+            TokenCacheItem tokenItem = new TokenCacheItem();
+            final String cacheKey = CacheKey.createCacheKey(this.getAuthority(), null, foci, true, uniqueUserId);
+            tokenItem = this.getCache().getItem(cacheKey);
+            
+            if (tokenItem == null) {
+                Logger.i(TAG, "Cannot find the FoCI token cache item for this userID", "");
+                return null;
+            }
+
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(BlobContainer.class, new BlobContainerAdapter())
+                    .create();
+            final BlobContainer blobContainer = new BlobContainer(tokenItem);
+            Logger.i(TAG, "family refresh token cache item:" + blobContainer.toString(), "");
+            String json = gson.toJson(blobContainer);
+            return json;           
+        }      
+    }
+    
+    protected void deserialize(String serializedBlob) throws AuthenticationException {
+        if (StringExtensions.IsNullOrBlank(serializedBlob)) {
+            throw new IllegalArgumentException("serializedBlob");
+        }
+            
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_IMPORT,"Failed to import the serialized blob because broker is enabled.");
```

Comment:
Also if we need to throw, then the else block is not needed

---
In reply to: [62591073](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/609#discussion_r62591073) [](ancestors = 62591073)